py/plugins/ollama/src/genkit/plugins/ollama/plugin_api.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2025 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Ollama Plugin for Genkit.
+"""
+
+import logging
+from typing import Callable
+
+import ollama as ollama_api
+
+from genkit.plugins.ollama.models import OllamaPluginParams
+from genkit.plugins.ollama.utils import (
+    register_ollama_embedder,
+    register_ollama_model,
+)
+from genkit.veneer import Genkit
+
+LOG = logging.getLogger(__name__)
+
+
+def Ollama(plugin_params: OllamaPluginParams) -> Callable[[Genkit], None]:
```

Comment:
just thinking here -- in JS we have a helper function for defining plugins
https://github.com/firebase/genkit/blob/ee4afba9ee1d7cd0d888829cfd3b312bd27fc826/js/genkit/src/plugin.ts#L31

wondering it's worth doing something similar.
The benefit of have it is that it can encapsulate genkit level plugin handling logic, like if the plugin will need to be initialized lazily, etc. things like that can be tweaked later, perhaps avoiding breaking changes to the plugins.

Reply:
> just thinking here -- in JS we have a helper function for defining plugins
> 
> https://github.com/firebase/genkit/blob/ee4afba9ee1d7cd0d888829cfd3b312bd27fc826/js/genkit/src/plugin.ts#L31
> 
> wondering it's worth doing something similar. The benefit of have it is that it can encapsulate genkit level plugin handling logic, like if the plugin will need to be initialized lazily, etc. things like that can be tweaked later, perhaps avoiding breaking changes to the plugins.

Yeah, we also though of making a separate interface for the plugin, so gonna handle it first, and then refactor this PR